packages/components/src/internal/components/ontology/OntologyConceptAnnotation.tsx

```diff
@@ -0,0 +1,144 @@
+    const [showSelectModal, setShowSelectModal] = useState<boolean>();
+    const [showConceptModal, setShowConceptModal] = useState<boolean>();
+    const isFieldLocked = useMemo(() => isFieldFullyLocked(lockType), [lockType]);
+    const title = 'Select Concept';
+
+    const toggleSelectModal = useCallback(() => {
+        setShowSelectModal(!showSelectModal);
+    }, [showSelectModal, setShowSelectModal]);
+
+    const toggleConceptModal = useCallback(() => {
+        setShowConceptModal(!showConceptModal);
+    }, [showConceptModal, setShowConceptModal]);
+
+    const onApply = useCallback(
+        (selectedConcept: ConceptModel) => {
+            onChange(id, selectedConcept?.code);
+            setShowSelectModal(false);
+        },
+        [onChange, id, setShowSelectModal]
+    );
+
+    return (
+        <>
+            <div className="domain-field-label">
+                <DomainFieldLabel label="Concept Annotation" helpTipBody={getOntologyConceptAnnotationHelpTipBody()} />
+            </div>
+            <table className="domain-annotation-table">
+                <tbody>
+                    <tr>
+                        <td>
+                            <Button
+                                className="domain-validation-button"
+                                name={createFormInputName(DOMAIN_FIELD_ONTOLOGY_PRINCIPAL_CONCEPT)}
+                                id={id}
+                                disabled={isFieldLocked}
+                                onClick={toggleSelectModal}
+                            >
+                                {title}
+                            </Button>
+                        </td>
+                        {!principalConceptCode && (
+                            <td className="content">
+                                <span className="domain-text-label">None Set</span>
+                            </td>
+                        )}
+                        {principalConceptCode && (
+                            <>
+                                {!isFieldLocked && (
+                                    <td className="content">
+                                        <a className="domain-validator-link" onClick={() => onApply(undefined)}>
+                                            <i className="fa fa-remove" />
+                                        </a>
+                                    </td>
+                                )}
+                                <td className="content">
+                                    <a
+                                        className="domain-validator-link domain-annotation-item"
+                                        onClick={toggleConceptModal}
+                                    >
+                                        {concept?.getDisplayLabel() ?? principalConceptCode}
+                                    </a>
+                                </td>
+                            </>
+                        )}
```

Comment:
We could but in discussion with other React devs, we have started going with this approach where instead of using the if/else syntax (i.e. `true ? <component1/> : <component2/>`) or going with `renderX` functions, we are going with this type of syntax which is hopefully a little more readible.